catalog/CatalogRelation.cpp

```diff
@@ -116,13 +111,26 @@ CatalogRelation::CatalogRelation(const serialization::CatalogRelation &proto)
 #endif
   }
 
+  // Deserializing the index scheme defined for the relation, if any.
+  if (proto.has_index_scheme()) {
+    index_scheme_.reset(IndexScheme::ReconstructFromProto(proto.index_scheme()));
+    // Ensure that indices present in the block layout are the same as in the index scheme.
+    const std::size_t num_indices_expected = index_scheme_->getNumIndices();
+    const std::size_t num_indices_checked = proto.default_layout().index_description_size();
+    DCHECK(num_indices_expected == num_indices_checked);
+    for (int i = 0; i < proto.default_layout().index_description_size(); ++i) {
```

Comment:
We could use `num_indices_checked`, instead of `proto.default_layout().index_description_size()`.